### PR TITLE
Enhancement/340/configurable screenshot folder

### PR DESCRIPTION
**Pull request by cliffle**

### CODE REVIEW

File: packages/e2e/e2e-suite/sakuli.io/typed-check.ts [hunk 1]
```diff
@@ -11,9 +11,11 @@
         await _wait( 5000, async () => (await driver.getAllWindowHandles()).length === 2);
         await driver.switchTo().window((await driver.getAllWindowHandles())[1]);
         await _highlight(_code("npm init"));
-        testCase.endOfStep("Find npm init code sample");
+        await driver.close();
+        await driver.switchTo().window((await driver.getAllWindowHandles())[0]);
```
Comment (svettwer) [inline packages/e2e/e2e-suite/sakuli.io/typed-check.ts:15]: Why is this required now?
The driver should be closed by Sakuli. A manual driver close should not be required at any time.

File: packages/e2e/e2e-suite/sakuli.io/check.js [hunk 2]
```diff
@@ -1,27 +1,22 @@
...
-        await _pageIsStable(5000, 500);
-        await _assertEqual("M-Package", await _getSelectedText(_select("powermail_field_kundenwunsch")));
-        await testCase.endOfStep("Contact form clicked");
+        await _click(_link("Docs"));
+        testCase.endOfStep("Navigate to docs");
+        await _wait( 5000, async () => (await driver.getAllWindowHandles()).length === 2);
+        await driver.switchTo().window((await driver.getAllWindowHandles())[1]);
+        await _highlight(_code("npm init"));
+        await driver.close();
+        await driver.switchTo().window((await driver.getAllWindowHandles())[0]);
```
Comment (svettwer) [inline packages/e2e/e2e-suite/sakuli.io/check.js:15]: Same here with the driver (see typed check)

File: packages/sakuli-legacy/src/context/common/test-case/test-case.class.spec.ts [hunk 3]
```diff
@@ -241,6 +241,25 @@ describe("TestCase", () => {
...
+            legacyProps.screenshotDir = tmpdir();
+            project = mockPartial<Project>({
+                objectFactory: jest.fn().mockReturnValue(legacyProps)
+            });
+            const SUT = createTestCaseClass(testExecutionContext, project, testFolder);
+            const tc = new SUT("testId", 0, 0);
+            const testError = new Error("testError");
+
+            // WHEN
+            await tc.handleException(testError);
```
Comment (svettwer) [inline packages/sakuli-legacy/src/context/common/test-case/test-case.class.spec.ts:257]: This code fragment duplicates Line 222-234 now. This implies the test you wrote is already existing but missing a condition. I would suggest to add the `expect(ScreenApi.takeScreenshotWithTimestamp).toBeCalledWith(`${tmpdir()}/UNKNOWN_TESTSUITE_testcase_1/error_UNKNOWN_TESTSUITE_testcase_1`);` condition to the upper test case and alter the name instead of duplicating test code.

**Comment by svettwer** [comment]
There is one open point for me concerning the manual driver shutdown. Maybe we can discuss this briefly via zoom.
In addition, I've found a code duplication that should be resolved.

File: packages/e2e/e2e-suite/sakuli.io/typed-check.ts (hunk 1 above)
Comment (cliffle) [reply packages/e2e/e2e-suite/sakuli.io/typed-check.ts:15]: This is required because we now have two test cases opening a new tab.
Instead of using `CTRL/CMD+W` (we need the differentiation because Travis uses both Linux and macOS) to close a tab the webdriver already provides the [functionality](https://www.selenium.dev/documentation/en/webdriver/browser_manipulation/#closing-a-window-or-tab) with `driver.close()`. So with closing the tabs, it is no longer relevant in which order the test cases are run.

Comment (svettwer) [reply packages/e2e/e2e-suite/sakuli.io/typed-check.ts:15]: Okay, good to know. Thx for clarification. :+1:
